hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/AbstractTestS3AEncryption.java

```diff
@@ -79,7 +79,7 @@ protected void patchConfigurationEncryptionSettings(
   };
 
   protected void requireEncryptedFileSystem() {
-    skipIfEncryptionTestsDisabled(getFileSystem().getConf());
+    skipIfEncryptionTestsDisabled(createConfiguration());
```

Comment:
works as long as there's no looking for any per-bucket property resolution. which is not the case here, right?

Reply:
good to know. we're just looking up `test.fs.s3a.encryption.enabled` here.

Am I right to assume pre-bucket configs are limited to `fs.s3a.*`?

Reply:
Confirming limited to `fs.s3a.*`.

https://github.com/apache/hadoop/blob/7fb9c306e22eee334667d4b761e3fd330f45cb01/hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java#L1130-L1132

Reply:
fs.s3a, and the process doesn't handle buckets with . in their name